Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -618,9 +618,9 @@ extension PurchasesOrchestrator: PaymentQueueWrapperDelegate {
         // when `StoreKit1Wrapper` is not initialized, which means that promoted purchases
         // need to be handled as a SK2 purchase.
         // This method converts the `SKPayment` into an SK2 purchase by fetching the product again.
-        assert(self.storeKit1Wrapper == nil, "This method should not be invoked if SK1 is enabled")
         if self.storeKit1Wrapper != nil {
             Logger.warn("Unexpectedly received PaymentQueueWrapperDelegate call with SK1 enabled")
+            assertionFailure("This method should not be invoked if SK1 is enabled")
```

Comment:
This is equivalent, but avoids duplicating the expectation in the `if` and the `assert`.